internal/serialization/default_compact_reader.go

```diff
@@ -71,71 +227,685 @@ func (r DefaultCompactReader) ReadString(fieldName string) *string {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadInt8Array()
+	}, pubserialization.FieldKindArrayOfInt8, pubserialization.FieldKindArrayOfNullableInt8, "Int8").([]int8)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfInt16(fieldName string) []int16 {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadInt16Array()
+	}, pubserialization.FieldKindArrayOfInt16, pubserialization.FieldKindArrayOfNullableInt16, "Int16").([]int16)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfInt32(fieldName string) []int32 {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadInt32Array()
+	}, pubserialization.FieldKindArrayOfInt32, pubserialization.FieldKindArrayOfNullableInt32, "Int32").([]int32)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfInt64(fieldName string) []int64 {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadInt64Array()
+	}, pubserialization.FieldKindArrayOfInt64, pubserialization.FieldKindArrayOfNullableInt64, "Int64").([]int64)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfFloat32(fieldName string) []float32 {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadFloat32Array()
+	}, pubserialization.FieldKindArrayOfFloat32, pubserialization.FieldKindArrayOfNullableFloat32, "Float32").([]float32)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfFloat64(fieldName string) []float64 {
+	return d.readArrayOfPrimitive(fieldName, func(inp *ObjectDataInput) interface{} {
+		return inp.ReadFloat64Array()
+	}, pubserialization.FieldKindArrayOfFloat64, pubserialization.FieldKindArrayOfNullableFloat64, "Float64").([]float64)
+}
+
+func (d *DefaultCompactReader) ReadArrayOfString(fieldName string) []*string {
+	var values []*string
+	d.readArrayOfVariableSize(fieldName, pubserialization.FieldKindArrayOfString, func(inp *ObjectDataInput, i int32, isNil bool) {
+		if isNil {
+			values[i] = nil
+		} else {
+			str := inp.ReadString()
+			values[i] = &str
+		}
+	}, func(i int) {
+		values = make([]*string, i)
+	})
+	return values
+}
+
+func (d *DefaultCompactReader) ReadArrayOfDecimal(fieldName string) []*types.Decimal {
+	var values []*types.Decimal
+	d.readArrayOfVariableSize(fieldName, pubserialization.FieldKindArrayOfDecimal, func(inp *ObjectDataInput, i int32, isNil bool) {
+		if isNil {
+			values[i] = nil
+		} else {
+			dec := ReadDecimal(inp)
+			values[i] = &dec
+		}
+	}, func(i int) {
+		values = make([]*types.Decimal, i)
+	})
+	return values
+}
+
+func (d *DefaultCompactReader) ReadArrayOfTime(fieldName string) []*types.LocalTime {
+	var values []*types.LocalTime
+	d.readArrayOfVariableSize(fieldName, pubserialization.FieldKindArrayOfTime, func(inp *ObjectDataInput, i int32, isNil bool) {
+		if isNil {
+			values[i] = nil
+		} else {
+			lt := types.LocalTime(ReadTime(inp))
+			values[i] = &lt
+		}
```

Comment:
```suggestion
		if !isNil {
			lt := types.LocalTime(ReadTime(inp))
			values[i] = &lt
		}
```
I don't think we need to explicitly assign nil. But if you think that's more readable I am not against it either. Just a suggestion :)

Reply:
done in 75ed99583c488633d8ae2ea784b890d66dc874ab